x-pack/plugins/uptime/public/components/monitor_management/monitor_config/monitor_name_location.tsx

```diff
@@ -52,6 +74,84 @@ export const MonitorNameAndLocation = ({ validate }: Props) => {
           onChange={(event) => setName(event.target.value)}
         />
       </EuiFormRow>
+      {/* Advanced options toggle */}
+      <EuiFormRow>
```

Comment:
The toggle and the content look large enough to split out to a 2nd component. 

That being said, it sounds like the suggestion from Andrew was to move this to Advanced Options. In that case, I would move it to the bottom of Advanced Options, and use the existing pattern with a `EuiDescribedFormGroup` with the description of the field to the left and the field to the right.

One thing to note is that `ICMP` doesn't currently have advanced options, and will need to be set up.

Reply:
Yep, once I move the component into the advanced options section for each monitor type, it can be simplified and I don't think it will need to be split into a different file